woocommerce/woocommerce-ios
# WIP – Xcode 14.0

Closed this because I wanted the branch name to be `xcode-14`, not `xcode-14-beta-1`. 

---

### Description

This is going to be a long running branch to (try to) stay on top of the Xcode 14 betas PRs.

I expected many of you will want to checkout this branch locally to play with it. As such, I think we should keep it alive by _merging `trunk`_ into it. But, I think once we'll officially switch to Xcode 14 stable, it'll make sense to do a big rebase so the history stays tidy.

I called the branch `xcode-14`. This might give us the option to keep the beta branch stable (🙃 🙂) and work on a new beta support without breaking the "stable" branch. That is, we might have an `xcode-14-beta-2` branch in the future.

**Ideally, we should implement as many fixes for errors and warnings found here as possible on `trunk`.** See:

- #7057 

### Testing instructions

Download Xcode 14 beta 1 and run this branch 😅 .

If you do, you might notice a bunch of:

> Cannot access property 'cancellableSiteID' with a non-sendable type 'AnyCancellable?' from non-isolated deinit; this is an error in Swift 6

I haven't looked into it yet, other than finding [this thread about it in the Swift forums](https://forums.swift.org/t/cannot-access-property-with-a-non-sendable-type-from-non-isolated-deinit/57877).

### Screenshots

<table>
<tr>
	<td><img width="1579" alt="Screen Shot 2022-06-09 at 7 37 25 pm" src="https://user-images.githubusercontent.com/1218433/172820518-957247b8-a71e-46f4-a998-cf329a256edd.png"></td>
	<td><img width="561" alt="Screen Shot 2022-06-09 at 7 44 16 pm" src="https://user-images.githubusercontent.com/1218433/172820652-e0618679-4289-455a-84b3-ed1855c705cc.png"></td>
</table>

---
